lib/tasks/accounts/update_adopter_status.rake

```diff
@@ -0,0 +1,36 @@
+namespace :accounts do
+  desc 'Update newly created field for adopter status from Salesforce'
+  # rake accounts:update_adopter_status
+  task update_adopter_status: [:environment] do
+    last_id = nil
+    loop do
+      contacts = OpenStax::Salesforce::Remote::Contact.select(
+        :id,
+        :adoption_status,
+        :accounts_uuid
+      )
+      .where("Accounts_UUID__c != null")
+      .order(:Id)
+      .limit(250)
+      contacts = contacts.where("Id > '#{last_id}'") unless last_id.nil?
+      contacts = contacts.to_a
+      last_id = contacts.last&.id
+
+      begin
+        user_by_salesforce_id = User.where(
+          salesforce_contact_id: contacts.map(&:id)
+        ).index_by(&:salesforce_contact_id)
+
+        updated_contacts = contacts.map do |contact|
+          user = user_by_salesforce_id[contact.id]
+          user.adopter_status = contact.adoption_status
+          user.save!
+        end
```

Comment:
I'd wrap each batch inside a separate transaction. I think that will improve performance (rather than using autocommit to automatically create a transaction for each save!).

Reply:
I think this is how it's done but let me know if this doesn't look like the correct way to do this.